src/test/java/htsjdk/variant/variantcontext/VariantContextBuilderTest.java

```diff
@@ -47,19 +55,207 @@ public void testAttributeResettingWorks(final boolean leaveModifyableAsIs) {
+    public void testFilterCanUseUnmodifyableSet() {
+        final Set<String> filters = new HashSet<>();
+        filters.add("TEST");
+        final VariantContextBuilder builder = new VariantContextBuilder("source", "contig", 1, 1, Arrays.asList(Tref, C)).filters(Collections.unmodifiableSet(filters));
+        builder.filter("CanIHazAFilter?");
+        final VariantContext vc1 = builder.make();
+
+        Assert.assertEquals(vc1.getFilters().size(),2);
+    }
+
+    @DataProvider
+    public static Object[][] illegalFilterStrings() {
+        return new Object[][]{
+                {"Tab\t"},
+                {"newLine\n"},
+                {"space "},
+                {"semicolon;"},
+                {"carriage return\r"}
+        };
+    }
+
+    @Test(dataProvider = "illegalFilterStrings",expectedExceptions = IllegalStateException.class)
+    public void testFilterCannotUseBadFilters(final String filter) {
+        final Set<String> filters = new HashSet<>();
+        filters.add(filter);
+        final VariantContextBuilder builder = new VariantContextBuilder("source", "contig", 1, 1, Arrays.asList(Tref, C)).filters(Collections.unmodifiableSet(filters));
+        final VariantContext vc1 = builder.make();
+
+        //shouldn't have gotten here
+        Assert.fail("a bad filter should have not been permitted: '" + filter + "'");
+    }
+
+
+    @Test(dataProvider = "BuilderSchemes")
+    public void testAllelesUnaffectedByClonedVariants(final VCBuilderScheme builderScheme) {
+        final VariantContextBuilder builder = new VariantContextBuilder("source", "contig", 1, 1, Arrays.asList(Tref, C)).filter("TEST");
+        final VariantContext ignored = builder.make();
+        final VariantContext vc2 = builderScheme.getOtherBuilder(builder, ignored).alleles(Collections.singleton(Aref)).make();
+
+        final VariantContext vc1 = builder.make();
+
+        if (builderScheme == VCBuilderScheme.SAME) {
+            Assert.assertEquals(vc2.getAlleles(), vc1.getAlleles(), "The two lists of alleles should be the same");
+        } else {
+            Assert.assertNotEquals(vc2.getAlleles(), vc1.getAlleles(), "The two lists of alleles should be different");
+        }
+    }
+
+    @Test(dataProvider = "BuilderSchemes")
+    public void testGenotypesUnaffectedByClonedVariants(final VCBuilderScheme builderScheme) {
+        if (builderScheme == VCBuilderScheme.NEW_ON_VC) {
+            return;
+        }
+
+        final VariantContextBuilder builder = new VariantContextBuilder("source", "contig", 1, 1, Arrays.asList(Tref, C, G)).filter("TEST");
+
+        final Genotype sample1 = GenotypeBuilder.create("sample1", Arrays.asList(Tref, C));
+        final Genotype sample2 = GenotypeBuilder.create("sample2", Arrays.asList(Tref, G));
+
+        GenotypesContext gc = GenotypesContext.create(sample1);
+        builder.genotypes(gc);
+        try {
+            gc.add(sample2);
+        } catch (IllegalAccessError e) {
+            // nice work...
```

Comment:
Wow, I didn't realize that this throws `IllegalAccessError`.  That's a subclass of `LinkageError`.  We should absolutely not be throwing or catching that here.   

Could you change it to `IllegalStateException` instead?

Reply:
no.. the issue is that the sample map is immutable, so when I try to add a sample to it I get this exception (as I should!)

Reply:
oh. I see what you mean. sure.